core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -96,14 +79,12 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
             filter.toXContent(builder, params);
         }
         builder.endArray();
-        if (queryName != null) {
-            builder.field("_name", queryName);
-        }
+        printBoostAndQueryName(builder);
```

Comment:
yes there is a very good reason for this, that boost and queryName don't always have the same position in the json, so trying to generify it is going to make things even messier I think, especially given that the parsers cannot be generified at all.